src/FirstClaim/FirstClaim.sol

```diff
@@ -0,0 +1,214 @@
+import {IGrant} from "../IGrant.sol";
+
+contract FirstClaim is Ownable2Step {
+    ////////////////////////////////////////////////////////////////
+    //                           ERRORS                           //
+    ////////////////////////////////////////////////////////////////
+
+    /// @notice Error that is thrown if the caller is not allowed to call the function
+    error OnlyAllowedCaller();
+
+    /// @notice Error that is thrown if a proposed configuration address is the zero address
+    error ZeroAddress();
+
+    /// @notice Error that is thrown if the grant amount is too large
+    error GrantAmountTooLarge();
+
+    /// @notice Error that is thrown if the owner tries to renounce ownership
+    error CannotRenounceOwnership();
+
+    ////////////////////////////////////////////////////////////////
+    //                           EVENTS                           //
+    ////////////////////////////////////////////////////////////////
+
+    /// @notice Event emitted when a caller is added
+    event CallerAdded(address indexed caller);
+
+    /// @notice Event emitted when a caller is removed
+    event CallerRemoved(address indexed caller);
+
+    /// @notice Event emitted when the AllowanceModule is set
+    event AllowanceModuleSet(address indexed allowanceModule);
+
+    /// @notice Event emitted when the WLD token is set
+    event WldTokenSet(address indexed wldToken);
+
+    /// @notice Event emitted when the Holder is set
+    event HolderSet(address indexed holder);
+
+    /// @notice Event emitted when the RecurringGrantDrop is set
+    event RecurringGrantDropSet(address indexed recurringGrantDrop);
+
+    /// @notice Event emitted when a first claim has been made
+    event FirstClaimClaimed(uint256 grantId, address indexed receiver, uint256 amount, uint256 currentGrantAmount);
+
+    ////////////////////////////////////////////////////////////////
+    ///                      CONFIG STORAGE                      ///
+    ////////////////////////////////////////////////////////////////
+
+    /// @notice address of the Safe Allowance Module
+    AllowanceModule public ALLOWANCE_MODULE;
+
+    /// @notice Worldcoin token address
+    address public WLD_TOKEN;
+
+    /// @notice BVI Safe that grants allowances to this contract
+    GnosisSafe public HOLDER;
+
+    /// @notice Recurring Grant Drop contract
+    IRecurringGrantDrop public RECURRING_GRANT_DROP;
+
+    /// @notice addresses that can call the claim function
+    mapping(address => bool) public allowedCallers;
+
+    ////////////////////////////////////////////////////////////////
+    ///                       CONSTRUCTOR                        ///
+    ////////////////////////////////////////////////////////////////
+
+    constructor(
+        address _allowanceModuleAddress,
+        address _wldToken,
+        address _holder,
+        address _recurringGrantDrop
+    ) Ownable(msg.sender) {
```

Comment:
q: do you maybe want to pass an owner here, so we don't need to do the accept ownership?